### DISCUSSION TITLE
Payload dist output is to heavy

### DISCUSSION THREAD

**Discussion by TarasOP** [payloadcms/payload#5400] [General]
### Describe the Bug

I want to deploy the payload app to the amplify hosting. But the output is greater 220 MB, so the error occurs. 

### To Reproduce

You can create a payload project and run the build command. Then run `du -h -d1 . | sort -h`

### Payload Version

2.11.2

### Adapters and Plugins

@payloadcms/plugin-form-builder, @payloadcms/db-mongodb

**Comment by DanRibbens** [comment]
Yes, in the next version of payload, v3, the dependencies are being reduced drastically. That is a focus we have been working on and will be in beta in a few weeks. 

Usually the dependencies for a Payload project are okay with 1GB of memory. Make sure you don't have any extras in your dependencies that could be moved to devDependencies.

If you post your package.json we can try and point out improvements. 
I'm converting this to a discussion in the meantime.

**Reply by TarasOP** [reply]
Hello, thanks for instant reply 

```{
  "name": "corporate-website-cms",
  "description": "A blank template to get started with Payload",
  "version": "1.0.0",
  "main": "dist/server.js",
  "license": "MIT",
  "scripts": {
    "dev:aws:login": "aws sso login --profile corporate-website-development",
    "upgrade-packages": "yarn upgrade-interactive --latest",
    "dev": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts nodemon",
    "build:payload": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload build",
    "build:server": "tsc",
    "build": "yarn copyfiles && yarn build:payload && yarn build:server",
    "serve": "cross-env PAYLOAD_CONFIG_PATH=dist/payload.config.js NODE_ENV=production node dist/server.js",
    "copyfiles": "copyfiles -u 1 \"src/**/*.{html,css,scss,ttf,woff,woff2,eot,svg,jpg,png}\" dist/",
    "generate:types": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload generate:types",
    "generate:graphQLSchema": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload generate:graphQLSchema",
    "payload": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload",
    "postbuild": "chmod +x bin/postbuild.sh && ./bin/postbuild.sh"
  },
  "dependencies": {
    "@aws-sdk/client-s3": "^3.536.0",
    "@payloadcms/bundler-webpack": "^1.0.0",
    "@payloadcms/db-mongodb": "^1.0.0",
    "@payloadcms/plugin-form-builder": "^1.2.1",
    "@payloadcms/richtext-slate": "^1.0.0",
    "cross-env": "^7.0.3",
    "dotenv": "^16.4.5",
    "express": "^4.18.3",
    "payload": "^2.11.2"
  },
  "devDependencies": {
    "@types/express": "^4.17.9",
    "copyfiles": "^2.4.1",
    "nodemon": "^3.1.0",
    "ts-node": "^10.9.2",
    "typescript": "^5.4.2"
  }
}
```

**Reply by TarasOP** [reply]
the output of `du -h -d1 . | sort -h `

4.0K    ./bin
112K    ./dist
124K    ./src
1.8M    ./.git
7.0M    ./build
408M    ./node_modules
416M    ./.amplify-hosting
834M    .

**Reply by TarasOP** [reply]
.amplify-hosting should be less 220MB 
without node_modules included it will not work 
I tried to install only production dependencies, but the output was about  ~390MB

**Reply by DanRibbens** [reply]
Yeah, that isn't going to work under that small of memory constraints until we get to v3 and we can keep that target in mind.

This is a node ecosystem problem, we're doing everything we can do minimize dependencies going forward but there might not be a lot we can do in the moment as I see it.

**Reply by TarasOP** [reply]
I will keep in mind that. Thanks a lot

**Reply by DanRibbens** [reply]
~390MB is less than I expected, but there isn't much more that can be done to get this down until v3 I don't think.
